src/py42/services/userriskprofile.py

```diff
@@ -0,0 +1,177 @@
+from py42.exceptions import Py42BadRequestError
+from py42.exceptions import Py42CloudAliasCharacterLimitExceededError
+from py42.exceptions import Py42CloudAliasLimitExceededError
+from py42.exceptions import Py42Error
+from py42.exceptions import Py42NotFoundError
+from py42.exceptions import Py42UserRiskProfileNotFound
+from py42.services import BaseService
+from py42.services.util import get_all_pages
+
+
+class UserRiskProfileService(BaseService):
+
+    _uri_prefix = "/v1/user-risk-profiles"
+
+    def get_by_id(self, user_id):
+        uri = f"{self._uri_prefix}/{user_id}"
+        try:
+            return self._connection.get(uri)
+        # catch not found error
+        except Py42NotFoundError as err:
+            raise Py42UserRiskProfileNotFound(err, user_id)
+
+    def get_by_username(self, username):
+        user_id = None
+        generator = self.get_all()
+
+        # get the first page of user profiles
+        for page in generator:
+            for user in page.data["userRiskProfiles"]:
+                if user["username"] == username:
+                    user_id = user["userId"]
+                    break
+
+        try:
+            return self.get_by_id(user_id)
+        except Py42NotFoundError as err:
+            raise Py42UserRiskProfileNotFound(err, username, identifier="username")
+
+    def update(self, user_id, start_date=None, end_date=None, notes=None, paths=None):
```

Comment:
we should also accept a `datetime` object for the date params, like we do elsewhere: https://github.com/code42/py42/blob/main/src/py42/services/detectionlists/departing_employee.py#L50

Reply:
Hmmm we handle `datetime.datetime` objects but not `datetime.date` ones, I'm gonna do both